provider_client.go

```diff
@@ -412,6 +412,10 @@ func (client *ProviderClient) doRequest(method, url string, options *RequestOpts
 		}
 	}
 
+	for _, v := range options.OmitHeaders {
+		req.Header.Del(v)
```

Comment:
Is there any concern that the headers from `client.AuthenticatedHeaders()` will override this setting (line 420-422)?

Reply:
Before commit 29581a2fc7076bae722d4b681f9cfaa467aa3e09, `client.AuthenticatedHeaders()` is prior to `options.MoreHeaders`. In my case, `options.OmitHeaders` makes up for the previous `options.MoreHeaders` and should inherit the original order

Reply:
yeah, we expected that something could be broken, see https://github.com/gophercloud/gophercloud/issues/2215#issuecomment-921114591

Reply:
I'm not sure that something _is_ broken yet, though. I think there needs to be some investigation and comparison to @freedywu's use-case that is triggering this and the basic acceptance test that passes without changes required.

Reply:
> Is there any concern that the headers from client.AuthenticatedHeaders() will override this setting

I _think_ it should be okay given that the order of the existing code has `MoreHeaders` first adding an empty `X-Auth-Token` and then `client.AuthenticatedHeaders` adds headers on top of it. Definitely warrants running this PR through the Keystone suite of acceptance tests, though.